### ISSUE TITLE
Table creation via REST API does not enforce requested order of clustering columns

### ISSUE THREAD

**Issue by jeffreyscarpenter** [stargate/stargate#1841]
The create table operation on the REST API appears to create clustering keys in the order the columns are declared, not in the order specified in the primary key declaration.

For example:

```
POST /v2/schemas/keyspaces/<your keyspace here>/tables

{
    "name": "some_table",
    "columnDefinitions": [
        {
            "name": "pk1",
            "typeDefinition": "text"
        },
        {
            "name": "ck1",
            "typeDefinition": "text"
        },
        {
            "name": "ck2",
            "typeDefinition": "text"
        }
    ],
    "primaryKey": {
        "partitionKey": [
            "pk1"
        ],
        "clusteringKey": [
            "ck2", "ck1"
        ]
    },
    "tableOptions": {
        "defaultTimeToLive": 0,
        "clusteringExpression": [
            {
                "column": "ck1",
                "order": "ASC"
            },
            {
                "column": "ck2",
                "order": "ASC"
            }
        ]
    }
}
```

When reading the table definition back via CQL, the primary key should reflect the order specified in the clustering key: `pk1, ck2, ck1`, but in fact it will read `pk1, ck1, ck2` (ordering of the clustering columns reversed).

**Comment by tatu-at-datastax** [comment]
Ok, neither Stargate V1 nor Stargate V2 table creation code seems to reorder column definition to be sorted same as primary key (partition, clustering keys) columns. So I think the problem would also affect partition keys

**Comment by mark-stargate** [comment]
Is this fixed by #1842?

**Comment by jeffreyscarpenter** [comment]
> Is this fixed by #1842?

No we didn't add a fix yet, I just created a test case to verify there was a bug.

**Comment by tatu-at-datastax** [comment]
Fix for SGv1/REST via #1842 (in code review); will next work on #1856 which fixes SGv2/REST

**Comment by tatu-at-datastax** [comment]
Included in release 1.0.57 / 2.0.0-ALPHA-11.
